### ISSUE TITLE
SDR not changing between epoches

### ISSUE THREAD

**Issue by RavenV-tiff** [ZFTurbo/Music-Source-Separation-Training#5]
This may not be an actual issue, but my training method. Would like thoughts on my application.

I'm using the starting weight from mdx23c and the vocals config file.
I have ~300 tracks that I'm training with and 5 for testing. Which takes about 30 minutes per epoch on my 12 GB card.
I used to have the datasets split 80% (215 tracks) training and 20% (85 tracks) testing which would take over an hour just for the testing part.

SDR for the 80-20 dataset were Avg 13.8
SDR for the reduced testing are Avg 14.9

However, after the first training epoch any subsequent epoch the SDR stagnates in both training methods. And since my card has 12GB I can only afford to batch train on a single song, not the 6 for the project configured 48 GB memory cards. This is the only change I made to the configure file.

`audio:
  chunk_size: 261120
  dim_f: 4096
  dim_t: 256
  hop_length: 1024
  n_fft: 8192
  num_channels: 2
  sample_rate: 44100
  min_mean_abs: 0.001

model:
  act: gelu
  bottleneck_factor: 4
  growth: 128
  norm: InstanceNorm
  num_blocks_per_scale: 2
  num_channels: 128
  num_scales: 5
  num_subbands: 4
  scale:
  - 2
  - 2

training:
  batch_size: 1
  gradient_accumulation_steps: 1
  grad_clip: 0
  instruments:
  - vocals
  - other
  lr: 9.0e-05
  patience: 2
  reduce_factor: 0.95
  target_instrument: null
  num_epochs: 1000
  num_steps: 1000
  augmentation: false # enable augmentations by audiomentations and pedalboard
  augmentation_type: simple1
  use_mp3_compress: false # Deprecated
  augmentation_mix: true # Mix several stems of the same type with some probability
  augmentation_loudness: true # randomly change loudness of each stem
  augmentation_loudness_type: 1 # Type 1 or 2
  augmentation_loudness_min: 0.5
  augmentation_loudness_max: 1.5
  q: 0.95
  coarse_loss_clip: true
  ema_momentum: 0.999
  optimizer: adam
  other_fix: true # it's needed for checking on multisong dataset if other is actually instrumental

inference:
  batch_size: 1
  dim_t: 256
  num_overlap: 4`

What might explain the stagnating SDR? Could the batch size configuration be to blame or testing size? Would restarting the training after epoch 0 be advisable since it does improve the SDR after the first epoch, but perhaps it's overtraining then?

**Comment by ZFTurbo** [comment]
If SDR reduced it usually means learning rate is too high. Also it's not so easy to improve SDR of this vocal model because SDR is very high initially.
